doc/connectivity/bluetooth/api/mesh/shell.rst

```diff
@@ -327,44 +326,80 @@ The Proxy Client model is an optional mesh subsystem that can be enabled through
 Configuration Client model
 ==========================
 
-The Configuration Client model is an optional mesh subsystem that can be enabled through the :kconfig:option:`CONFIG_BT_MESH_CFG_CLI` configuration option. If included, the Bluetooth mesh shell module instantiates a Configuration Client model for configuring itself and other nodes in the mesh network.
+The Configuration Client model is an optional mesh subsystem that can be enabled through the :kconfig:option:`CONFIG_BT_MESH_CFG_CLI` configuration option. This is implemented as a separate module (``mdl_cli``) alongside the ``mesh`` shell instance. This module will work on any instance of the Configuration Client model if the mentioned shell configuration options is enabled, and as long as the Configuration Client model is present in the model composition of the application. This shell module can be used for configuring itself and other nodes in the mesh network.
 
-The Configuration Client uses the general messages parameters set by ``mesh dst`` and ``mesh netidx`` to target specific nodes. When the Bluetooth mesh shell node is provisioned, the Configuration Client model targets itself by default. When another node has been provisioned by the Bluetooth mesh shell, the Configuration Client model targets the new node. The Configuration Client always sends messages using the Device key bound to the destination address, so it will only be able to configure itself and mesh nodes it provisioned.
+The module use ``mdl_cfg target set`` to target specific nodes, and ``mdl_cfg target get`` to see the current target. In most common use-cases the Configuration Client will be dependent on the provisioning features and the Configuration database to be fully functional. The Configuration Client always sends messages using the Device key bound to the destination address, so it will only be able to configure itself and mesh nodes it provisioned. The following steps is a example of how you can set up a device to start using the Configuration Client commands:
 
-``mesh timeout [timeout in seconds]``
--------------------------------------
+* Initialize the client node (``mesh init``).
+* Create the CDB (``mesh cdb create``).
+* Provision the local device (``mesh provision-local``).
+* Target the local node Configuration Server instance (``mdl_cfg target set <local_addr> <netidx>``).
+* Monitor the composition data of the local node (``mdl_cfg get-comp``).
+* Configure the local node as desired with the Configuration Client commands.
+* Provision other devices (``mesh beacon-listen``) (``mesh pb-adv``) (``mesh pb-gatt``).
```

Comment:
Well, it could but the devices the shell cli module can provision is not restricted to only other devices with this shell module, so I thought it might be a little bit misleading.

Reply:
Up to you. However, configuration module uses device key for app encryption\decryption. Only provisioner knows this. Naturally provisioner owns the config client. Otherwise we have to have some API to share device key.

Reply:
I think I mention a bit above in the docs that the cfg client in most common cases will be dependent on the provisioner feature